### ISSUE TITLE
when run i got this error without add any code for cloud_firestore

### ISSUE THREAD

**Issue by MTEV** [firebase/flutterfire#10833]
**when run i got this error without add any code for cloud_firestore**

```
Launching lib\main.dart on sdk gphone x86 in debug mode... C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:13: error: cannot find symbol import com.google.firebase.firestore.AggregateQuery; ^ symbol: class AggregateQuery location: package com.google.firebase.firestore C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:14: error: cannot find symbol import com.google.firebase.firestore.AggregateQuerySnapshot; ^ symbol: class AggregateQuerySnapshot location: package com.google.firebase.firestore C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:15: error: cannot find symbol import com.google.firebase.firestore.AggregateSource; ^ symbol: class AggregateSource location: package com.google.firebase.firestore C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\streamhandler\TransactionStreamHandler.java:19: error: cannot find symbol import com.google.firebase.firestore.TransactionOptions; ^ symbol: class TransactionOptions location: package com.google.firebase.firestore C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestoreMessageCodec.java:412: error: where(Filter) is not public in Query; cannot be accessed from outside package query = query.where(filter); ^ C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:536: error: cannot find symbol AggregateQuery aggregateQuery = query.count(); ^ symbol: class AggregateQuery location: class FlutterFirebaseFirestorePlugin C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:536: error: cannot find symbol AggregateQuery aggregateQuery = query.count(); ^ symbol: method count() location: variable query of type Query C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:537: error: cannot find symbol AggregateQuerySnapshot aggregateQuerySnapshot = ^ symbol: class AggregateQuerySnapshot location: class FlutterFirebaseFirestorePlugin C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestorePlugin.java:538: error: cannot find symbol Tasks.await(aggregateQuery.get(AggregateSource.SERVER)); ^ symbol: variable AggregateSource location: class FlutterFirebaseFirestorePlugin C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\streamhandler\TransactionStreamHandler.java:72: error: package TransactionOptions does not exist new TransactionOptions.Builder().setMaxAttempts(maxAttempts).build(), ^ Note: C:\Users\AppData\Local\Pub\Cache\hosted\pub.dev\cloud_firestore-4.5.2\android\src\main\java\io\flutter\plugins\firebase\firestore\FlutterFirebaseFirestoreMessageCodec.java uses unchecked or unsafe operations. Note: Recompile with -Xlint:unchecked for details. 10 errors

FAILURE: Build failed with an exception.

What went wrong: Execution failed for task ':cloud_firestore:compileDebugJavaWithJavac'.
Compilation failed; see the compiler error output for details.

Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

Get more help at https://help.gradle.org/
BUILD FAILED in 7s Exception: Gradle task assembleDebug failed with exit code 1 Exited (sigterm)

```
**minSdkVersion 21
my pubspec.yaml**

intl: ^0.18.0 shared_preferences: ^2.1.0

firebase_core: ^2.10.0

firebase_auth: ^4.4.2

google_sign_in: ^6.1.0

cloud_firestore: ^4.5.2

**any solution**

**Comment by darshankawar** [comment]
@MTEV 
Can you provide `flutter doctor -v` ? 
Also, try `flutter pub cache repair` followed by `flutter pub get` to see if it helps or not.

**Comment by MTEV** [comment]
flutter pub cache repair nothing  solve  same issue 

and -pubspec.yaml 
------------------------------------------------------------
name: mynew
description: A new Flutter project.

# The following line prevents the package from being accidentally published to
# pub.dev using `flutter pub publish`. This is preferred for private packages.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev

# The following defines the version and build number for your application.
# A version number is three numbers separated by dots, like 1.2.43
# followed by an optional build number separated by a +.
# Both the version and the builder number may be overridden in flutter
# build by specifying --build-name and --build-number, respectively.
# In Android, build-name is used as versionName while build-number used as versionCode.
# Read more about Android versioning at https://developer.android.com/studio/publish/versioning
# In iOS, build-name is used as CFBundleShortVersionString while build-number is used as CFBundleVersion.
# Read more about iOS versioning at
# https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
# In Windows, build-name is used as the major, minor, and patch parts
# of the product and file versions while build-number is used as the build suffix.
version: 2.4.0+14

environment:
  sdk: '>=2.18.1 <3.0.0'

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter
  

  carousel_slider: ^4.2.1
  http: ^0.13.5
  date_field: ^3.0.0
  sign_button: ^2.0.3
  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  flutter_login_facebook: ^1.7.0
  syncfusion_flutter_gauges: ^21.1.35
  material_dialogs: ^1.1.4
  animate_gradient: ^0.0.2
  validators: ^3.0.0
  awesome_dialog: ^3.0.2
 
  custom_radio_grouped_button: any
  # intl: 
  shared_preferences: ^2.1.0
  # firebase_core: any
  # firebase_auth: any
  # google_sign_in: ^6.1.0
  # cloud_firestore: ^4.5.2
  firebase_core: ^2.10.0
  firebase_storage: ^11.1.1
  #cloud_firestore: ^4.5.2
  # firebase_messaging: ^9.0.0
  firebase_auth: ^4.2.2
  google_sign_in: ^6.1.0
  firebase_core_platform_interface: ^4.6.0
  cloud_firestore: ^4.5.2
 
 

dev_dependencies:
  flutter_launcher_icons: ^0.13.1
  change_app_package_name: ^1.1.0
flutter_icons:
  android: "launcher_icon"
  ios: true
  image_path: "images/logoapp.png"
  min_sdk_android: 21

  flutter_test:
    sdk: flutter

  # The "flutter_lints" package below contains a set of recommended lints to
  # encourage good coding practices. The lint set provided by the package is
  # activated in the `analysis_options.yaml` file located at the root of your
  # package. See that file for information about deactivating specific lint
  # rules and activating additional ones.
  flutter_lints: ^2.0.0

# For information on the generic Dart part of this file, see the
# following page: https://dart.dev/tools/pub/pubspec

# The following section is specific to Flutter packages.
flutter:

  # The following line ensures that the Material Icons font is
  # included with your application, so that you can use the icons in
  # the material Icons class.
  uses-material-design: true

  # To add assets to your application, add an assets section, like this:
  # assets:
  #   - images/a_dot_burr.jpeg
  #   - images/a_dot_ham.jpeg
  assets:
    - images/intro_bg.jpg
    - images/intro_bgb.jpg
    - images/logoapp.png
    - images/assets/logo_footer.png
    - images/assets/logo_head.png
    - images/assets/logo.png
    - images/assets/google.png
    - images/assets/facebook.png
    - images/assets/user_password.png
    - images/assets/icon_eye_close.png
    - images/assets/icon_eye_open.png
    - images/assets/apple.png
    - images/assets/icon_password.png
    - assets/logo_footer.png
    - assets/logo_head.png
    - assets/logo.png
    - assets/google.png
    - assets/facebook.png
    - assets/user_password.png
    - assets/icon_eye_close.png
    - assets/icon_eye_open.png
    - assets/apple.png
    - assets/icon_user.png
    - assets/icon_password.png
    - assets/alogoapp.png
    #- assets/background.png


  # An image asset can refer to one or more resolution-specific "variants", see
  # https://flutter.dev/assets-and-images/#resolution-aware

  # For details regarding adding assets from package dependencies, see
  # https://flutter.dev/assets-and-images/#from-packages

  # To add custom fonts to your application, add a fonts section here,
  # in this "flutter" section. Each entry in this list should have a
  # "family" key with the font family name, and a "fonts" key with a
  # list giving the asset and other descriptors for the font. For
  # example:
  fonts:
    - family: Lobster-Regular
      fonts:
        - asset: fonts/Lobster-Regular.ttf
  #       - asset: fonts/Schyler-Italic.ttf
  #         style: italic
  #   - family: Trajan Pro
  #     fonts:
  #       - asset: fonts/TrajanPro.ttf
  #       - asset: fonts/TrajanPro_Bold.ttf
  #         weight: 700
  #
  # For details regarding fonts from package dependencies,
  # see https://flutter.dev/custom-fonts/#from-packages
# dependency_overrides:
#   intl: ^0.18.1

**Comment by MTEV** [comment]
> @MTEV Can you provide `flutter doctor -v` ? Also, try `flutter pub cache repair` followed by `flutter pub get` to see if it helps or not.
not working

**Comment by darshankawar** [comment]
Can you try by creating a new Flutter project, add required flutterfire plugins like core, auth and firestore and try to run and see if you still get the same error or not ? That way you can check if any other plugin you might be using could be causing the conflict or not.

**Comment by MTEV** [comment]
same issue

**Comment by MTEV** [comment]
> Can you try by creating a new Flutter project, add required flutterfire plugins like core, auth and firestore and try to run and see if you still get the same error or not ? That way you can check if any other plugin you might be using could be causing the conflict or not.

same issue

**Comment by MTEV** [comment]
can you try create new project and  just add    cloud_firestore: ^4.5.2
and telll me  if working  
just write 
flutter run

**Comment by MTEV** [comment]
> Can you try by creating a new Flutter project, add required flutterfire plugins like core, auth and firestore and try to run and see if you still get the same error or not ? That way you can check if any other plugin you might be using could be causing the conflict or not.

can you try create new project and just add cloud_firestore: ^4.5.2
and telll me if working
just write
flutter run

**Comment by darshankawar** [comment]
I tried as above and was able to run the app properly without the reported error:

```
Building with Flutter multidex support enabled.
Running Gradle task 'assembleDebug'...                             10.2s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app-debug.apk...           4.3s
Syncing files to device Android SDK built for x86...                99ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

💪 Running with sound null safety 💪

An Observatory debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:62895/frowBbZKIuo=/
The Flutter DevTools debugger and profiler on Android SDK built for x86 is available at: http://127.0.0.1:9100?uri=http://127.0.0.1:62895/frowBbZKIuo=/


```

Maybe try switching to latest master and run again. Also check what gradle build and Android SDK version you are using that might be leading to the error.
